src/main/kotlin/no/nav/familie/ks/sak/kjerne/arbeidsfordeling/ArbeidsfordelingService.kt

```diff
@@ -0,0 +1,151 @@
+    private val personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository,
+    private val integrasjonClient: IntegrasjonClient,
+    private val personOpplysningerService: PersonOpplysningerService,
+    private val oppgaveService: OppgaveService,
+    private val loggService: LoggService
+) {
+    fun fastsettBehandledeEnhet(behandling: Behandling, sisteVedtattBehandling: Behandling? = null) {
+        val aktivArbeidsfordelingPåBehandling =
+            arbeidsfordelingPåBehandlingRepository.finnArbeidsfordelingPåBehandling(behandling.id)
+
+        val oppdatertArbeidsfordelingPåBehandling: ArbeidsfordelingPåBehandling = if (behandling.erSatsendring()) {
+            fastsettBehandledeEnhetPåSatsendringsbehandling(
+                behandling,
+                sisteVedtattBehandling,
+                aktivArbeidsfordelingPåBehandling
+            )
+        } else {
+            val arbeidsfordelingsenhet = hentArbeidsfordelingsenhet(behandling)
+            when (aktivArbeidsfordelingPåBehandling) {
+                null -> arbeidsfordelingPåBehandlingRepository.save(
+                    ArbeidsfordelingPåBehandling(
+                        behandlingId = behandling.id,
+                        behandlendeEnhetId = arbeidsfordelingsenhet.enhetId,
+                        behandlendeEnhetNavn = arbeidsfordelingsenhet.enhetNavn
+                    )
+                )
+                else -> {
+                    if (!aktivArbeidsfordelingPåBehandling.manueltOverstyrt &&
+                        (aktivArbeidsfordelingPåBehandling.behandlendeEnhetId != arbeidsfordelingsenhet.enhetId)
+                    ) {
+                        aktivArbeidsfordelingPåBehandling.also {
+                            it.behandlendeEnhetId = arbeidsfordelingsenhet.enhetId
+                            it.behandlendeEnhetNavn = arbeidsfordelingsenhet.enhetNavn
+                        }
+                        arbeidsfordelingPåBehandlingRepository.save(aktivArbeidsfordelingPåBehandling)
+                    }
+                    aktivArbeidsfordelingPåBehandling
+                }
+            }
+        }
+        postFastsattBehandlendeEnhet(
+            behandling = behandling,
+            aktivArbeidsfordelingPåBehandling = aktivArbeidsfordelingPåBehandling,
+            oppdatertArbeidsfordelingPåBehandling = oppdatertArbeidsfordelingPåBehandling,
+            manuellOppdatering = false
+        )
+    }
+
+    fun hentArbeidsfordelingsenhet(behandling: Behandling): Arbeidsfordelingsenhet {
+        val søker = identMedAdressebeskyttelse(behandling.fagsak.aktør)
+        val personopplysningGrunnlag = personopplysningGrunnlagRepository.findByBehandlingAndAktiv(behandling.id)
+
+        val personer = when (personopplysningGrunnlag) {
+            null -> listOf(søker)
+            else -> personopplysningGrunnlag.barna.map { barn -> identMedAdressebeskyttelse(barn.aktør) }.plus(søker)
+        }
```

Comment:
Variende bruk av null checker her. Hva synes du er best @haldersaikat?
Ser at vi bruker when for å nullchecke her, men vanligvis hadde vi jo brukt noe sånt:

`val personer = personopplysningGrunnlag?.let { it.barna.map { barn -> identMedAdressebeskyttelse(barn.aktør) }.plus(søker) } ?: listOf(søker)
`

Reply:
Ja, enig. Jeg synes det er bedre å bruke sånn  -> val personer = personopplysningGrunnlag?.let { itbarna.map { barn -> identMedAdressebeskyttelse(barn.aktør) }.plus(søker) }

Reply:
Even shorter:
```
val personer = personopplysningGrunnlag?.barna?.map { identMedAdressebeskyttelse(it.aktør) }?.plus(søker)
?: listOf(søker)
```

Reply:
oi, kopierte feil, men har gjort akkurat det samme